src/modules/navigator/land.cpp

```diff
@@ -57,8 +57,18 @@ Land::on_activation()
 
 	/* convert mission item to current setpoint */
 	struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
-	pos_sp_triplet->previous.valid = false;
-	mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
+
+	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+		mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
+		pos_sp_triplet->previous.valid = false;
+
+	} else {
```

Comment:
I would rather have `if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) { // calculate breaking stop` and "else do the usual stuff". Otherwise you'll also compute the breaking stop for the rovers/boats/...